File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        for(auto& ch:s){
            // if(ch=='(' || ch=='{' || ch=='[') stack.push(ch);
            if(ch==')' && stack.size() && stack.top()=='(') stack.pop();
            else if(ch=='}' && stack.size() && stack.top()=='{') stack.pop();
            else if(ch==']' && stack.size() && stack.top()=='[') stack.pop();
            else stack.push(ch);
            // }
        }
        return stack.size()==0;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def validPop(self,topch,ch): 
        if (topch=='(' and ch==')' ) or (topch=='[' and ch==']') or (topch=='{' and ch=='}'):
            return True 
        return False
    def isValid(self, s: str) -> bool:
        st = [] 
        for ch in s: 
            if ch == '(' or ch=='[' or ch=='{':
                st.append(ch) 
            else: 
                if st and self.validPop(st[-1],ch):
                    st.pop() 
                else: 
                    st.append(ch)
        
        return len(st)==0